Directly compress mip maps from Basis Universal

fire opened the pull request:
Part of: https://github.com/godotengine/godot-proposals/issues/4271

Discussed with @reduz:

Only support basisu with uastc.

It has been veto'd to not include all possible platforms at the same time. Do not store all of etc2, dxt5, bptc at the same time due to excess file size.

---

To increase efficiency enable compressed mip maps from Basis Universal.
To improve quality enable UASTC RDO compression.
To keep compatibility use the first mip of the previous internal Godot format.

This does a trick where I use only the first image from now on to read its mipmaps.

----

Part of the V-Sekai group. https://github.com/v-sekai

akien-mga commented on editor/import/resource_importer_texture.cpp:
```diff
@@ -380,7 +376,8 @@ void ResourceImporterTexture::_save_ctex(const Ref<Image> &p_image, const String
 	Ref<Image> image = p_image->duplicate();
 
-	if (((p_compress_mode == COMPRESS_BASIS_UNIVERSAL) || (p_compress_mode == COMPRESS_VRAM_COMPRESSED && p_force_po2_for_compressed)) && p_mipmaps) {
+	if ((((p_compress_mode == COMPRESS_BASIS_UNIVERSAL) && p_force_po2_for_compressed) || (p_compress_mode == COMPRESS_VRAM_COMPRESSED && p_force_po2_for_compressed)) &&
+			p_mipmaps) {
```
```suggestion
	if (p_force_po2_for_compressed && p_mipmaps && ((p_compress_mode == COMPRESS_BASIS_UNIVERSAL) || (p_compress_mode == COMPRESS_VRAM_COMPRESSED))) {
```
Equivalent?

akien-mga commented on modules/basis_universal/register_types.cpp:
```diff
@@ -52,44 +54,47 @@ enum BasisDecompressFormat {
-		//params.m_quality_level = 0;
-		//params.m_disable_hierarchical_endpoint_codebooks = true;
-		//params.m_no_selector_rdo = true;
+		params.m_quality_level = 0;
```
You're setting it twice to different values.

fire replied:
Resolving.

fire replied:
Thanks, you found the cause of the lower than expected performance! Thanks.

akien-mga reviewed (commented):
Only gave it a very cursory check, would need in-depth review by @reduz. But the changes seem sensible, if BU can now support mipmaps properly we should use it.

fire committed:
Improvements to Basis Universal.

Enable compressed mip maps from Basis Universal for faster compressions.

Increase the quality to avoid corruption.

To keep compatibility use the first mip of the previous internal Godot format.

fire commented:
@lyuma since you worked on the mipmaps code, can you review that area?

fire commented:
Can be archived. I'll will merge it with https://github.com/godotengine/godot/pull/62499

fire commented:
tl;dr

This pr does three things.

1. use one mipmap call to basisu instead of several
2. Discussed maybe separating basisu high quality and basisu fast
3. basisu high quality uses uastc and lowest quality
4. basisu fast uses etc1s and highest possible quality

fire commented:
Superseded by: https://github.com/godotengine/godot/pull/62499
